test/test_nestedtensor.py

```diff
@@ -96,15 +96,23 @@ def noncontiguous_to_padded_tensor(input, shape=None):
 # Helper function to generate a random nested tensor
 
 
-def random_nt(device, dtype, num_tensors, max_dims, min_dims=None, layout=torch.strided):
+def random_nt(device, dtype, num_tensors, max_dims, min_dims=None, layout=torch.strided, must_non_empty=True):
     if min_dims is None:
         min_dims = tuple([0] * len(max_dims))
-    ts1 = []
-    for _ in range(num_tensors):
-        tensor_dims = tuple([torch.randint(low=min_dim, high=max_dim, size=(1,)).item()
-                            for (min_dim, max_dim) in zip(min_dims, max_dims)])
-        t1 = torch.randn(tensor_dims, device=device, dtype=dtype)
-        ts1.append(t1)
+    has_non_empty = False
+    while not has_non_empty:
+        # Repeat until we have at least one non-empty tensor
```

Comment:
technically this could go on forever :) although it becomes vanishingly unlikely

is it simple enough to refactor this so that we don't have a non-deterministic length loop?

Reply:
Yeah this is a good call out, a little more work, but its definitely better this way!